spring-cloud-contract-verifier/src/main/groovy/org/springframework/cloud/contract/verifier/builder/JsonBodyVerificationBuilder.java

```diff
@@ -123,6 +131,20 @@ Object addJsonResponseBodyCheck(BlockBuilder bb, Object convertedResponseBody,
 		return convertedResponseBody;
 	}
 
+	private boolean mapOrListBodyReducedToEmpty(Object originalBody,
+			Object convertedBody) {
+		int origSize = originalBody instanceof Map ? ((Map) originalBody).size() : -1;
```

Comment:
Well, this addition is, umm how do you say.... gross. Open to suggestions here. What I want to say represent is "is the object a list or map whose size was non-zero before the conversion but is zero after the conversion"

Reply:
I get the intent and the solution looks comprehensible and that's all that matters IMO